lib/spack/spack/test/config.py

```diff
@@ -1132,3 +1132,14 @@ def test_single_file_scope_cache_clearing(env_yaml):
     after = scope.get_section('config')
     assert after
     assert before == after
+
+
+def test_internal_config_cache():
+    config = spack.config.Configuration()
+    config.push_scope(spack.config.InternalConfigScope('internal', {
+        'config': {
+            'build_jobs': 10
+        }
+    }))
+    config.clear_caches()
```

Comment:
I would add a comment on what we're going to test for the next person reading this, for instance:
```suggestion
    # An InternalConfigScope object is constructed from data that is already
    # in memory, therefore it doesn't have any cache to clear. Here we ensure
    # that calling the clear method is a no-op.
    internal_scope.clear()
```